lightning/src/ln/channel.rs

```diff
@@ -5557,7 +5573,7 @@ impl<SP: Deref> Channel<SP> where
 	/// [`ChannelMonitorUpdate`] will be returned).
 	pub fn get_shutdown(&mut self, signer_provider: &SP, their_features: &InitFeatures,
 		target_feerate_sats_per_kw: Option<u32>, override_shutdown_script: Option<ShutdownScript>)
-	-> Result<(msgs::Shutdown, Option<ChannelMonitorUpdate>, Vec<(HTLCSource, PaymentHash)>), APIError>
+	-> Result<(msgs::Shutdown, Option<ChannelMonitorUpdate>, Vec<(HTLCSource, PaymentHash)>, Option<ShutdownResult>), APIError>
```

Comment:
I wonder if these `Option<ChannelMonitorUpdate>, Vec<(HTLCSource, PaymentHash)>` can be collapsed into `Option<ShutdownResult>` although the use cases are slightly different.

Reply:
Right, that was my main hesitation. It seems possible to branch on the `update_id` of the `ChannelMonitorUpdate` being equal to `CLOSED_CHANNEL_UPDATE_ID`, but might not be something we want to unify.